src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -85,9 +85,22 @@ class IntelliJPlugin implements Plugin<Project> {
                                                     @NotNull IntelliJPluginExtension extension) {
         LOG.info("Configuring IntelliJ IDEA dependency")
         def resolver = new IdeaDependencyManager(extension.intellijRepo ?: DEFAULT_INTELLIJ_REPO)
-        def ideaDependency = resolver.resolve(project, extension.version, extension.type, extension.downloadSources)
-        if (ideaDependency == null) {
-            throw new BuildException("Failed to resolve IntelliJ IDEA ${extension.version}", null)
+        def ideaDependency
+        if (extension.localPath != null) {
+            if (extension.version != null) {
+                LOG.warn("Both `localPath` and `version` specified, second would be ignored")
+            }
+            LOG.info("Using path to locally installed IDE: '${extension.localPath}'")
+            ideaDependency = resolver.resolveLocal(project, extension.localPath, extension.type)
+            if (ideaDependency == null) {
```

Comment:
Cannot be null. Moreover, the place in unreachable: you throw a BuildExpection in case of invalid path